pretyping/typing.ml

```diff
@@ -136,6 +136,12 @@ let is_correct_arity env sigma c pj ind specif params =
           srec (push_rel (LocalAssum (na1,a1)) env) sigma t ar'
       end
     | Sort s, [] ->
+        let sigma = match ESorts.kind sigma s with
+        | QSort (_, u) ->
+          (* Arbitrarily set the return sort to Type *)
+          Evd.set_eq_sort env sigma s (ESorts.make (Sorts.sort_of_univ u))
```

Comment:
Shouldn't we have some API like `Evd.collapse_sort_variable` instead of using with set_eq_sort?